nemo/collections/asr/modules/audio_modules.py

```diff
@@ -0,0 +1,526 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Optional
+
+import torch
+
+from nemo.collections.asr.parts.utils.audio_utils import db2mag
+from nemo.core.classes import NeuralModule, typecheck
+from nemo.core.neural_types import AudioSignal, FloatType, LengthsType, NeuralType, SpectrogramType
+from nemo.utils.decorators import experimental
+
+try:
+    import torchaudio
+
+    HAVE_TORCHAUDIO = True
+except ModuleNotFoundError:
+    HAVE_TORCHAUDIO = False
+
+
+__all__ = [
+    'AudioToSpectrogram',
+    'SpectrogramToAudio',
+    'MaskEstimatorRNN',
+    'MaskReferenceChannel',
+    'MaskBasedBeamformer',
+]
+
+
+@experimental
+class AudioToSpectrogram(NeuralModule):
```

Comment:
this is probably a replication of  https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/modules/audio_preprocessing.py#L95. it would just be adding one or two args to return intermediate STFT from AudioToMelSpectrogramPreprocessor or sub-classing it.

Reply:
It is precisely the reason that `AudioToMelSpectrogramPreprocessor`

Reply:
This class should be `AudioToMultiChannelMelSpectrogramProcessor` - long name, but much more descriptive of what it actually does.

In this sense, i do think it is fine to not subclass the original one, but i agree with Krishna that it will degrade user experience who have grown accustomed to the generic AudioToMelSpectrogramProcessor. 

You don't have to modify the internals if you subclass do you? Anyway, this is fine for now, we can improve it internally in the future if needed.

Reply:
created a future issue.